src/inference/session_context.py

```
@@ -727,7 +736,23 @@ def cleanup_old_sessions(self, max_age_days: int) -> int:
-            deleted = conn.execute("DELETE FROM sessions WHERE updated_at < ?", (cutoff,)).rowcount
+            # 삭제 대상 session_id 목록 먼저 조회 (metadata 테이블에는 FK CASCADE가 없음)
+            cursor = conn.execute(
+                "SELECT session_id FROM sessions WHERE updated_at < ?",
+                (cutoff,),
+            )
+            expired_ids = [row["session_id"] for row in cursor.fetchall()]
+
+            if expired_ids:
+                placeholders = ",".join("?" * len(expired_ids))
+                conn.execute(
+                    f"DELETE FROM metadata WHERE owner_type='session' AND owner_id IN ({placeholders})",
```

Ruff (0.15.9)

[error] Possible SQL injection vector through string-based query construction

(S608)